mp/src/game/server/momentum/mom_triggers.cpp

```diff
@@ -1573,7 +1579,21 @@ bool CNoGrenadesZone::IsInsideNoGrenadesZone(CBaseEntity *pOther)
             if (!pSticky)
                 return true; // This is a rocket in a nonade zone and should fizzle
 
-            return !pNoGrenadeZone->m_bAirborneOnly || !pSticky->DidHitWorld();
+            switch (pNoGrenadeZone->m_iStickyFizzleType)
+            {
+                case 0:
+	                return pPlayer->m_nButtons & IN_ATTACK2 && pSticky->IsArmed() && (!pNoGrenadeZone->m_bAirborneOnly || !pSticky->DidHitWorld()); // player must det for fizzle
+                case 1:
+                {
+                    return !pNoGrenadeZone->m_bAirborneOnly && pSticky->DidHitWorld() // fizzle if stick on ground and not airborne
+                            ||
+                            pNoGrenadeZone->m_bAirborneOnly && (pPlayer->m_nButtons & IN_ATTACK2 && pSticky->IsArmed() && pSticky->DidHitWorld());
+                            // if the mapper sets it for case 1 but enables airborne only, the sticky will be allowed to det improperly
+                            // this second check makes it fall back to case 0
+                }
+                default:
+                    return true; // fizzle if we're in at all
```

Comment:
Since we remove the `Touch` override in the stickybomb, this won't get called. 

This fizzle on touch can be handled in an `OnStartTouch` override in this nogrenades class.